Src/PChecker/CheckerCore/Specifications/Monitor.cs

```diff
@@ -177,6 +187,17 @@ internal void Initialize(ControlledRuntime runtime)
         {
             Runtime = runtime;
         }
+        
+        /// <summary>
+        /// Checks if the event is null, and calls <see cref="Monitor.RaiseEvent"/>.
+        /// </summary>
+        public void TryRaiseEvent(Event ev, object payload = null)
+        {
+            Assert(!(ev is DefaultEvent), "Monitor cannot raise a null event");
+            var oneArgConstructor = ev.GetType().GetConstructors().First(x => x.GetParameters().Length > 0);
```

Comment:
Do we need this part? Can we not set the `payload` field in the `ev` in the generated code and then just call `RaiseEvent` directly.